lib/config.js

```diff
@@ -756,6 +756,9 @@ util.parseFile = function(fullFilename) {
     fileContent = fileContent.replace(/^\uFEFF/, '');
   }
   catch (e2) {
+    if (e2.code !== 'ENOENT') {
+      throw new Error('Config file ' + fullFilename + ' cannot be read');
+    }
```

Comment:
I want to be sure I understand why this change is correct. Before parseFile is called, we will already making a "readdir" call and found which files exist. So, if we are parsing a file and it doesn't exist a moment after it did exist, that's unexpected so we know throw, right?

This a breaking change since we used to return `null` it would be good to document in the exception in the JSDoc comment above the function. 

Also, the line below "return null; // file doesn't exist", but that comment would no longer be accurate, because the "does not exist" case should be covered by ENOENT above.

Reply:
No, this is to count for cases in which `parseFile` is called from somewhere else, which is not `loadFileConfigs`, since these still happen. This is so we won't make a breaking change :)

Thing is, we got rid of `statsSync` so now we only wanna throw an exception only if it's anything other than "file don't exists", because we don't care about that. That's the official way suggested by NodeJS.

Reply:
> Also, the line below "return null; // file doesn't exist", but that comment would no longer be accurate, because the "does not exist" case should be covered by ENOENT above.

Notice we're check if it is **not** `ENOENT`, then we throw an exception.
We simply ignore cases in which the file doesn't exists.